From: Alon Bar-Lev <alon.barlev@gmail.com>

Current implementation allows the kernel to receive up to 255 characters from
the bootloader.  While the boot protocol allows greater buffers to be sent.

In current environment, the command-line is used in order to specify many
values, including suspend/resume, module arguments, splash, initramfs and
more.  

255 characters are not enough anymore. 

After edd issue was fixed, and dynammic kernel command-line patch was
accepted, we can extend the COMMAND_LINE_SIZE without runtime memory
requirements.

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
Cc: Andi Kleen <ak@suse.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>

Automatically created from "patches.suse/i386-2048-byte-command-line.patch" by xen-port-patches.py
Acked-by: Jan Beulich <jbeulich@novell.com>

Index: head-2007-02-08/include/asm-i386/mach-xen/asm/setup.h
===================================================================
--- head-2007-02-08.orig/include/asm-i386/mach-xen/asm/setup.h	2007-02-09 09:23:00.000000000 +0100
+++ head-2007-02-08/include/asm-i386/mach-xen/asm/setup.h	2007-02-09 09:37:11.000000000 +0100
@@ -6,7 +6,7 @@
 #ifndef _i386_SETUP_H
 #define _i386_SETUP_H
 
-#define COMMAND_LINE_SIZE 256
+#define COMMAND_LINE_SIZE 2048
 
 #ifdef __KERNEL__
 #include <linux/pfn.h>
